File: services/order-service/src/routers/tables.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from shared.dependencies.database import get_db
from ..schemas import tables as schema
from ..controllers import tables as controller

router = APIRouter(prefix="/tables", tags=["tables"])


@router.post("/", response_model=schema.Table)
def create_table(request: schema.TableCreate, db: Session = Depends(get_db)):
    return controller.create(db=db, request=request)


@router.get("/", response_model=list[schema.Table])
def get_all_tables(db: Session = Depends(get_db)):
    return controller.read_all(db)


@router.get("/with-sessions", response_model=list[schema.TableWithSession])
def get_tables_with_sessions(db: Session = Depends(get_db)):
    """Get all tables with their current session ID (for floor view)"""
    return controller.get_tables_with_session_info(db)


@router.get("/available", response_model=list[schema.Table])
def get_available_tables(db: Session = Depends(get_db)):
    return controller.get_available_tables(db)


@router.get("/occupied", response_model=list[schema.Table])
def get_occupied_tables(db: Session = Depends(get_db)):
    return controller.get_occupied_tables(db)


@router.get("/{table_id}", response_model=schema.Table)
def get_table(table_id: int, db: Session = Depends(get_db)):
    return controller.read_one(db, table_id=table_id)


@router.get("/code/{table_code}", response_model=schema.Table)
def get_table_by_code(table_code: str, db: Session = Depends(get_db)):
    return controller.read_by_code(db, table_code=table_code)


@router.put("/{table_id}", response_model=schema.Table)
def update_table(table_id: int, request: schema.TableUpdate, db: Session = Depends(get_db)):
    return controller.update(db=db, table_id=table_id, request=request)


@router.delete("/{table_id}")
def delete_table(table_id: int, db: Session = Depends(get_db)):
    return controller.delete(db=db, table_id=table_id)

File: apps/pos-system/src/lib/api/endpoints/tables.ts
import { apiClient } from '../client.js';
import type { Table, TableWithSession, TableSeating } from '$lib/types/index.js';

export const tablesApi = {
    // Get all tables
    async getAllTables(): Promise<Table[]> {
        return apiClient.request<Table[]>('tables/');
    },

    // Get all tables with current session info (for floor view)
    async getTablesWithSessions(): Promise<TableWithSession[]> {
        return apiClient.request<TableWithSession[]>('tables/with-sessions');
    },

    // Get available tables
    async getAvailableTables(): Promise<Table[]> {
        return apiClient.request<Table[]>('tables/available');
    },

    // Get occupied tables
    async getOccupiedTables(): Promise<Table[]> {
        return apiClient.request<Table[]>('tables/occupied');
    },

    // Get table by ID
    async getTable(id: number): Promise<Table> {
        return apiClient.request<Table>(`tables/${id}`);
    },

    // Get table by code
    async getTableByCode(code: string): Promise<Table> {
        return apiClient.request<Table>(`tables/code/${code}`);
    }
};

export const tableSeatingsApi = {
    // Get all active seatings
    async getActiveSeatings(): Promise<TableSeating[]> {
        return apiClient.request<TableSeating[]>('seatings/active');
    },

    // Create new seating
    async createSeating(data: { 
        table_id: number; 
        assigned_server_id?: number;
        notes?: string;
    }): Promise<TableSeating> {
        return apiClient.request<TableSeating>('seatings/', {
            method: 'POST',
            body: JSON.stringify(data)
        });
    },

    // Close seating
    async closeSeating(seatingId: number): Promise<TableSeating> {
        return apiClient.request<TableSeating>(`seatings/${seatingId}/close`, {
            method: 'POST'
        });
    },

    // Update seating
    async updateSeating(seatingId: number, data: Partial<TableSeating>): Promise<TableSeating> {
        return apiClient.request<TableSeating>(`seatings/${seatingId}`, {
            method: 'PUT',
            body: JSON.stringify(data)
        });
    }
};